heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java

```diff
@@ -52,27 +53,35 @@ public static boolean isValidInstance(Resource instanceResources,
       return false;
     }
 
-    if (instanceResources.getRam() > maxContainerResources.getRam()) {
+    if (instanceResources.getRam() + (paddingPercentage * instanceResources.getRam()) / 100
+        > maxContainerResources.getRam()) {
       LOG.severe(String.format(
           "This instance requires containers of at least %d MB ram. The current max container"
               + "size is %d MB",
-          instanceResources.getRam(), maxContainerResources.getRam()));
+          instanceResources.getRam() + (paddingPercentage * instanceResources.getRam()) / 100,
```

Comment:
best to calculate the pad only once and use that in both the criteria and the logging. Same for below.